### PR TITLE
[v7r2] Use prompt_toolkit for reading passwords

### PR DESCRIPTION
**Pull request by chrisburr**
Extracted from #4997 

BEGINRELEASENOTES

*Core
CHANGE: Use prompt-toolkit for password input when generating proxies

ENDRELEASENOTES

### CODE REVIEW

**Review by fstagni — DISMISSED** [review]

**Comment by petricm** [comment]
ok, maybe I should have asked it here. Does this `dirac-proxy-init <<< $CERT_PASSPHRASE` work?

File: src/DIRAC/FrameworkSystem/Client/ProxyUpload.py [hunk 1]
```diff
@@ -129,11 +129,14 @@ def uploadProxy(params):
...
-      passwdPrompt = "Enter Certificate password:"
       if params.stdinPasswd:
         userPasswd = sys.stdin.readline().strip("\n")
       else:
-        userPasswd = getpass.getpass(passwdPrompt)
+        try:
+          userPasswd = prompt(u"Enter Certificate password: ", is_password=True)
+        except KeyboardInterrupt:
+          DIRAC.gLogger.error("Caught KeyboardInterrupt, exiting...")
+          sys.exit(1)
```
Comment (fstagni) [inline src/DIRAC/FrameworkSystem/Client/ProxyUpload.py:139]: Same here, just
```suggestion
          return S_ERROR("Caught KeyboardInterrupt, exiting...")
```

File: src/DIRAC/FrameworkSystem/Client/ProxyGeneration.py [hunk 2]
```diff
@@ -297,11 +297,14 @@ def generateProxy(params):
...
-    passwdPrompt = "Enter Certificate password:"
     if params.stdinPasswd:
       userPasswd = sys.stdin.readline().strip("\n")
     else:
-      userPasswd = getpass.getpass(passwdPrompt)
+      try:
+        userPasswd = prompt(u"Enter Certificate password: ", is_password=True)
+      except KeyboardInterrupt:
+        gLogger.error("Caught KeyboardInterrupt, exiting...")
+        sys.exit(1)
```
Comment (fstagni) [inline src/DIRAC/FrameworkSystem/Client/ProxyGeneration.py:307]: You should return `S_ERROR` here, and the caller will call `sys.exit(1)`
```suggestion
        return S_ERROR("Caught KeyboardInterrupt, exiting...")
```

**Review by fstagni — APPROVED** [review]

**Comment by petricm** [comment]
Do not release without DIRACOS v1r19

**Comment by chrisburr** [comment]
> Do not release without DIRACOS v1r19

A new release isn't needed as `prompt_toolkit` is already there (if it wasn't the integration tests would be broken).

**Comment by petricm** [comment]
ok then I stop the production of v1r19 if `prompt-toolkit==1.0.18` is ok.

**Review by atsareg — APPROVED** [review]

**Pull request merged**